src/dataflow/components/dataflow-program.tsx

```diff
@@ -220,4 +235,23 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
       (plotControl as any).update();
     }
   }
+
+  private updateGeneratorNode = (n: Node) => {
+    const generatorType = n.data.generatorType;
+    const frequency: any = n.data.frequency;
+    const amplitude: any = n.data.amplitude;
```

Comment:
Are these really `any`? Seems like we ought to be able to have better types at this point.

Reply:
My understanding is that the Rete node data is designed to have properties freely added to store node information.   Makes it tricky to get a type off of the data when it is read.

Reply:
But we know that `frequency` and `amplitude` are expected to be numeric in our code, so could we do something like:
```
const frequency = Number(n.data.frequency);
const amplitude = Number(n.data.amplitude);
```
and now we're dealing with typed values in our code, independent of what Rete is doing internally.

Reply:
Sounds good.  Fixed.